tests/e2e/tidbcluster/serial.go

```diff
@@ -761,23 +765,56 @@ var _ = ginkgo.Describe("[tidb-operator][Serial]", func() {
 			err = oa.WaitForTidbClusterReady(tc, 30*time.Minute, 15*time.Second)
 			framework.ExpectNoError(err, "Check TidbCluster error")
 			monitor := fixture.NewTidbMonitor("monitor", ns, tc, false, false)
+
+			// Replace Prometheus into Mock Prometheus
+			l := strings.Split(e2econfig.TestConfig.E2EImage, ":")
+			image := ""
+			for _, s := range l[:len(l)-1] {
+				image = fmt.Sprintf("%s%s:", image, s)
+			}
+			monitor.Spec.Prometheus.BaseImage = image[:len(image)-1]
+			monitor.Spec.Prometheus.Version = l[len(l)-1]
```

Comment:
this is tricky, my suggestion:

```
	a := "localhost:5000/test:abc"
	colonIdx := strings.LastIndexByte(a, ':')
	image := a[:colonIdx]
	tag := a[colonIdx+1:]
```

https://play.golang.org/p/_HdHxL-JI45

Reply:
updated.